velox/core/QueryCtx.h

```diff
@@ -125,15 +120,88 @@ class QueryCtx {
     return queryId_;
   }
 
-  void testingOverrideMemoryPool(std::shared_ptr<memory::MemoryPool> pool) {
-    pool_ = std::move(pool);
-  }
+  /// Checks if the associated query is under memory arbitration or not. The
+  /// function returns true if it is and set future which is fulfilled when the
+  /// the memory arbiration finishes.
+  bool checkArbitration(ContinueFuture* future);
```

Comment:
Not sure if it's a good idea to put so much control logic in a context class. Maybe we can create a concrete Query class?

Reply:
Yeah, we don't have query level abstraction but only has query context for now. We could consider to have query abstraction and move it to there if we have seen more needs like this to better organize the multi-tasks query execution in velox.